chapter8.adoc

```diff
@@ -1,37 +1,124 @@
+refers to the Debug specification cite:[ExtDbg], Trace specification
+cite:[ETrc], and the External debug security extension cite:[ExtDbgSec] for the
+description and behavior of controls outside the scope of this specification,
+but which interact with controls specified in this specification when one or
+more supervisor domains are used. Note that `Smsdedbg` and `Smsdetrc` may be
+used as independent extensions even when a single supervisor domain is being
+utilized on a platform.
 
-=== `Smsdedbg` interaction with external debug security controls (Informative)
+[NOTE]
+====
+The configuration for `msdcfg.sdedbgalw` and `msdcfg.sdetrcalw` is expected to
+be obtained from the manifest/configuration of the supervisor domain and
+reported in the attestation report for the supervisor domain and it's workloads.
+The RDSM should manage this supervisor domain configuration state for context
+switching using secure memory.
+====
+
+=== `Smsdedbg`: External Debug allowed control for Supervisor Domain
+
+When M-mode external debug is enabled, all supervisor domains may also be
+debugged by an external debugger irrespective of the configuration held in
+`msdcfg.sdedbgalw`.
+
+When M-mode external debug is disabled, whether execution at privilege modes
+less than `M-mode` may be debugged by an external debugger depends on the
+configuration held in `msdcfg.sdedbgalw`, as described below:
+
+When `msdcfg.sdedbgalw` = 0, external debug is disallowed for the supervisor
+domain. Abstract commands and halt request from the debug module are suppressed
+and stay pending while the supervisor domain is active, on a per-hart basis.
+Other debug operations (current or defined in the future) must similarly be kept
+pending while the supervisor domain is active on a hart. Triggers for the
+supervisor domain must be controlled by the RDSM to prevent any leakage of
+information across security domains.
+
+When `msdcfg.sdedbgalw` = 1 then external debug of privilege modes less than
+`M-mode` is allowed for such a supervisor domain on a per-hart basis. +
+
+To enforce the above controls specified by this extension, the following
+functional and security requirements must be met by the external secure debug
+system cite:[ExtDbgSec] to meet the security objectives of supervisor domain
+isolation:
+
+. The enable control for external debug driven by the external debug module is
+  expected to be established by the platform root-of-trust following RISC-V
+  Security Model recommendations SR_GEN_007 and SR_GEN_012. When the control is
+  in cleared state, the hart should not be able to enter external debug.
+. External debug must be able to transition the hart to Debug Mode and access
+  supervisor domain memory and state. In this context, "state" includes all
+  non-M-mode resources accessible per the Debug specification cite:[ExtDbg].
+. Triggers must match and fire on a hart only when a supervisor domain with this
+  configuration is active on a hart (and not in M-mode).
```

Comment:
This can also be removed since we already stated that Debug Mode cannot be entered. Firing or/not of triggers depends on context switching of the triggers by the RDSM.

Reply:
ack